core/unix/module_elf.h

```diff
@@ -143,7 +145,14 @@ elf_loader_map_file(elf_loader_t *elf, bool reachable);
 /* Maps in the PT_LOAD segments of an ELF file, returning the base.  Must be
  * called after reading program headers with elf_loader_read_phdrs() or the
  * elf_loader_read_headers() shortcut.  All image mappings are done via the
- * provided function pointers.
+ * provided function pointers.  If a remap_func is specified, it is used when
+ * we must unmap a certain part of a prior reserved address range and use it
+ * for another mapping; unlike unmap_func followed by map_func, remap_func
+ * is expected to do this atomically to mitigate risk of that region getting
+ * mmaped by another thread between the unmap and map events (i#7192). On
+ * Linux, this may be achieved if the map call uses MAP_FIXED which
```

Comment:
I think someone new reading this will think "well, Linux provides mremap, why aren't you just using that?"  Make it clear this needs to switch from non-file-backed to file-backed so it's not as simple as using mremap.